azure-kusto-data/azure/kusto/data/__init__.py

```diff
@@ -42,7 +42,7 @@ class _ValidKeywords(Enum):
 
     def __init__(self, connection_string):
         self._internal_dict = {}
-        if connection_string is not None and "=" not in connection_string.split(";")[0]:
+        if connection_string is not None and "=" not in connection_string.partition(";")[0]:
```

Comment:
Maybe change the first condition to:
" if connection_string "
This will check both that connection_string is not None or either an empty string.